Improve Following class documentation and method efficiency

obenland opened the pull request:
Fixes #

## Proposed changes:

This PR improves the Following class with better documentation and more efficient method implementations:

* **Remove unused import**: Removed unused `Activitypub\Http` import
* **Fix @see references**: Updated `@see` tags to use proper namespace format (`\Activitypub\follow` instead of `Activitypub\follow`)
* **Improve return type documentation**: Updated `follow()` method documentation to accurately reflect all possible return types
* **Optimize count methods**: Modified count methods (`count_following`, `count_pending`, `count_all`) to use minimal parameters (limit=1) instead of fetching all data when only the total count is needed
* **Fix documentation typo**: Corrected return type documentation in `get_all_with_count()` to reference 'following' instead of 'followers'

### Other information:

- [x] Have you written new tests for your changes, if applicable? - No new functionality, only optimizations and documentation improvements

## Testing instructions:

* Go to WordPress admin and navigate to ActivityPub settings
* Test following/unfollowing functionality to ensure it still works correctly
* Verify that follower counts are displayed correctly in the UI
* Check that performance is maintained or improved when viewing large follower lists

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch

#### Type

-   [x] Changed - for changes in existing functionality

#### Message

Improve Following class documentation and optimize count methods for better performance

</details>

obenland committed:
Improve Following class documentation and method efficiency

- Remove unused Http import
- Fix @see references to use proper namespace format (\Activitypub\follow)
- Update return type documentation for follow() method to be more accurate
- Optimize count methods to avoid unnecessary data retrieval by using minimal number parameter
- Fix return type documentation for get_all_with_count to correctly reference 'following' instead of 'followers'

matticbot committed:
Add changelog

obenland commented on includes/collection/class-following.php:
```diff
@@ -54,14 +53,14 @@ class Following {
 	 * @param int          $user_id The ID of the WordPress User.
 	 *
-	 * @return \WP_Post|\WP_Error The ID of the Actor or a WP_Error.
+	 * @return int|false|\WP_Post|\WP_Error The Outbox ID or false on failure, the Actor post or a WP_Error.
```
@pfefferle The possible return values of this method are a bit confusing. It's a WP_Error with an invalid actor, the outbox item ID on success and false on failure, and if we didn't get to adding it to the outbox, the actor post. Would it be possible to limit it to two types?

pfefferle replied:
should we do this as part of this PR or in a follow up?

can you file an issue if it should be a follow up?

obenland replied:
Created issue #2092 to track the return type simplification for the `Following::follow()` method

obenland committed:
Update return type annotations for follow and unfollow

Adjusted the PHPDoc return types for the follow and unfollow functions to more accurately reflect their possible return values. This improves code documentation and helps with static analysis.

obenland committed:
Merge branch 'fix/following-types' of https://github.com/Automattic/wordpress-activitypub into fix/following-types

obenland committed:
Pass limit parameter to get_followers_with_count

Updated count_followers to pass a limit of 1 to get_followers_with_count, ensuring the method signature is correctly followed and potentially improving performance.

obenland committed:
Merge branch 'trunk' into fix/following-types
